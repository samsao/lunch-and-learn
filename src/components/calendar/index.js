export { default as SpeakerView } from './SpeakerView';
export { default as TalkDetail } from './TalkDetail';
export { default as ExpandedTalkDetail } from './ExpandedTalkDetail';
export { default as DateListHeader } from './DateListHeader';
export { default as TalkView } from './TalkView';
