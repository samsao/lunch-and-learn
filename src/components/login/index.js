export { default as LoginForm } from './LoginForm';
export { default as LoginTitle } from './LoginTitle';
