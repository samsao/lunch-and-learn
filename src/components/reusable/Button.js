import React from 'react';
import { Text, TouchableOpacity, StyleSheet } from 'react-native';
import { Padding, Colors } from '@mobile/constants/style/';

const styles = StyleSheet.create({
  text: {
    padding: Padding.EXTRA_SMALL,
  },
});

export default class Button extends React.Component {
  props: {
    text: string,
    onButtonTap: () => void,
    isButtonActive: boolean,
  }

  styles = StyleSheet.create({
    container: {
      backgroundColor: this.props.isButtonActive ? Colors.GREEN : Colors.LIGHT_GREY,
      borderColor: this.props.isButtonActive ? Colors.GREEN : Colors.LIGHT_GREY,
      borderWidth: 5,
      borderRadius: 50,
      width: 150,
      alignItems: 'center',
    },
  })

  render() {
    return (
      <TouchableOpacity
        onPress={this.props.onButtonTap}
        style={this.styles.container}
      >
        <Text style={styles.text}>{this.props.text}</Text>
      </TouchableOpacity>
    );
  }
}
