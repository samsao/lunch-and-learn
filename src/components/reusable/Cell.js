import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Padding, Colors } from '@mobile/constants/style/';

const styles = StyleSheet.create({
  container: {
    borderWidth: 1,
    borderRadius: 2,
    borderColor: Colors.LIGHT_GREY,
    shadowColor: Colors.BLACK,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 2,
    elevation: 1,
    marginLeft: Padding.MEDIUM,
    marginRight: Padding.MEDIUM,
    marginTop: Padding.SMALL,
    marginBottom: Padding.EXTRA_SMALL,
    backgroundColor: Colors.WHITE,
  },
});

export default class Cell extends React.Component {
  props: {
    style?: {},
    children?: typeof React.Element,
  };

  static defaultProps = {
    children: null,
  }

  static defaultProps = {
    style: styles.container,
  };

  render() {
    return (
      <View style={this.props.style}>
        {this.props.children}
      </View>
    );
  }
}
