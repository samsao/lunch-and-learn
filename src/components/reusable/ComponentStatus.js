export const FAILURE = 'FAILURE';
export const SUCCESS = 'SUCCESS';
export const STATELESS = 'STATELESS';
