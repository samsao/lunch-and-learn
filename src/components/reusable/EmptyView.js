import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  view: {
    justifyContent: 'center',
    alignItems: 'center',
    height: '90%',
  },
});

export default class EmptyView extends React.Component {
  props: {
    text: string
  }

  render() {
    return (
      <View>
        <View style={styles.view}><Text>{this.props.text}</Text></View>
      </View>
    );
  }
}
