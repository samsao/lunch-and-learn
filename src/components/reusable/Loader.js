import React from 'react';
import { View, ActivityIndicator, StyleSheet } from 'react-native';
import { Colors } from '@mobile/constants/style/';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    zIndex: 100,
    position: 'absolute',
    width: '100%',
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Colors.TRANSPARENT,
  },
});

export default class Loader extends React.Component {
  props: {
    display: boolean
  }
  render() {
    const pointerEvent = this.props.display ? 'box-only' : 'none';
    return (
      <View
        style={styles.container}
        pointerEvents={pointerEvent}
      >
        {this.props.display ? <ActivityIndicator color={Colors.GREEN} /> : undefined}
      </View>
    );
  }

}
