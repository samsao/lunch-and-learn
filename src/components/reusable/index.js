export { default as Button } from './Button';
export { default as EmptyView } from './EmptyView';
export { default as Loader } from './Loader';
export { default as NavigationBar } from './NavigationBar';
export { default as Cell } from './Cell';
export { default as TextField } from './TextField';
export { default as Notification } from './Notification';
