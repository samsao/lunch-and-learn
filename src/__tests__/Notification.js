import 'react-native';
import React from 'react';

import { Notification } from '@mobile/components/reusable';
import renderer from 'react-test-renderer';
import "jest-snapshots-svg";

/* eslint-disable indent */
describe('Notification Component', () => {
  it('renders properly in success mode', () => {
    const tree = renderer.create(
      <Notification
        status={'SUCCESS'}
        text={'Succeeded'}
        display={true}
        duration={null}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toMatchSVGSnapshot(320, 568);
  });

  it('renders properly in failure mode', () => {
    const tree = renderer.create(
      <Notification
        status={'FAILURE'}
        text={'Failed'}
        display={true}
        duration={null}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toMatchSVGSnapshot(320, 568);
  });

  it('renders properly in stateless mode', () => {
    const tree = renderer.create(
      <Notification
        status={'STATELESS'}
        text={'Info message'}
        display={true}
        duration={null}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toMatchSVGSnapshot(320, 568);
  });
});
