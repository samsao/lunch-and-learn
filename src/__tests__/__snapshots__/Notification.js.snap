// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Component renders properly in failure mode 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FF1212",
      "height": 40,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFF",
        "fontSize": 14,
      }
    }
  >
    Failed
  </Text>
</View>
`;

exports[`Notification Component renders properly in stateless mode 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#DDD",
      "height": 40,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000",
        "fontSize": 14,
      }
    }
  >
    Info message
  </Text>
</View>
`;

exports[`Notification Component renders properly in success mode 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#00AE58",
      "height": 40,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFF",
        "fontSize": 14,
      }
    }
  >
    Succeeded
  </Text>
</View>
`;
